File: test_project/gen_code_lua/tbitem.lua
-- Table TbItem
local TbItem = {
    _dataMap = {},
    _dataList = {}
}

function TbItem.Load(dataList)
    TbItem._dataList = dataList
    for _, v in ipairs(dataList) do
        TbItem._dataMap[v.Id] = v
    end
end

function TbItem.Get(key)
    return TbItem._dataMap[key]
end

function TbItem.GetDataList()
    return TbItem._dataList
end

function TbItem.GetDataMap()
    return TbItem._dataMap
end

return TbItem

File: test_project/gen_code_java/tbitem.java
package demo;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class TbItem {
    private final Map<int, Item> _dataMap;
    private final List<Item> _dataList;

    public TbItem(List<Item> dataList) {
        this._dataList = dataList;
        this._dataMap = new HashMap<>();
        for (Item v : _dataList) {
            this._dataMap.put(v.Id, v);
        }
    }

    public Map<int, Item> getDataMap() {
        return _dataMap;
    }

    public List<Item> getDataList() {
        return _dataList;
    }

    public Item get(int key) {
        return _dataMap.get(key);
    }
}
